src/internal/partial_solution.rs

```diff
@@ -214,13 +219,14 @@ impl<P: Package, V: Version> PartialSolution<P, V> {
         &self,
         incompat: &Incompatibility<P, V>,
     ) -> (Assignment<P, V>, DecisionLevel, DecisionLevel) {
-        let (
-            AssignmentDepth {
-                decision_level: satisfier_level,
-                assignment: satisfier,
-            },
-            previous_assignments,
-        ) = Self::find_satisfier(incompat, self.history.as_slice())
+        let AssignmentHistory {
```

Comment:
These really should not be put together though in my opinion. They have no other reason to be together than being two outputs of a function.

Reply:
The logic inside the satisfier isn't something very stable either. so having the flexibility of easily changing the return types there is important.

Reply:
> The logic inside the satisfier isn't something very stable either. so having the flexibility of easily changing the return types there is important.

Doesn't having a grouping type make it more easily changeable? You could extend the struct/change inner fields without touching type signature.

> These really should not be put together though in my opinion. They have no other reason to be together than being two outputs of a function.

True, the purpose of the change was to provide some context on those return types.
Before:
```
 ) -> Option<(AssignmentDepth<P, V>, &'a [AssignmentDepth<P, V>])>
```
After:
```
 ) -> Option<AssignmentHistory<'a, P, V>>
```
I think looking at type signature of "before" version does not make it clear why those are separate and why AssignmentDepth is not inside the slice, and what that slice means.

Would you be okay with making this a type alias if you think a struct definition doesn't pull it's weight here?
I would like having one of those as a return type.

Reply:
> Doesn't having a grouping type make it more easily changeable?

Yeah you're right on this one XD.

> I think looking at type signature of "before" version does not make it clear why those are separate and why AssignmentDepth is not inside the slice, and what that slice means.

I'll try to think a bit more about that return type and come back with some ideas (or not)